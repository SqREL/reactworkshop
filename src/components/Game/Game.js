import "./Game.scss";

import React from "react";
import PropTypes from "prop-types";

const Game = props => {
  return (
    <div className="c-game">
      <div className="c-game__img-wrapper">
        <img src={props.image} className="c-game__img" />
      </div>
      <div className="c-game__content">
        <div className="c-game__name">{props.name}</div>
        <div className="c-game__price">
          Price: {props.price > 0 ? props.price : "Free"}
        </div>
        <button className="o-button c-game__add-button" onClick={props.onAdd}>
          Add to cart
        </button>
      </div>
    </div>
  );
};

Game.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  image: PropTypes.string.isRequired,
  onAdd: PropTypes.func.isRequired
};

export default Game;
