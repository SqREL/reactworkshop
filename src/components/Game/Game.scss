@import '../../assets/styles/abstracts/variables';

.c-game {
    background: #fff;
    box-shadow: 0 2px 4px 0 rgba(0,0,0,0.07);

    &__img {
        width: 100%;
    }

    &__content {
        padding: $gutter-width/2;
    }

    &__name {
        font-weight: 900;
    }

    &__add-button {
        margin-top: 10px;
    }
}
