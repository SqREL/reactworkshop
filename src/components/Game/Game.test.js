import React from 'react';
import Game from "./Game"
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'


const props = {
  name: 'test game name',
  price: 1488,
  image: 'http://google.com',
  onAdd: jest.fn()
}

describe('Game', () => {
  it('handles onAdd', () => {
    const wrapper = shallow(<Game {...props}/>)

    wrapper.find('button').simulate('click')
    expect(props.onAdd).toBeCalled();
  })
  it('renders without crashing', () => {
    expect(toJson(shallow(<Game {...props}/>))).toMatchSnapshot()
  })
})

describe('Prce free', () => {
  const npr = {...props, price: 0}
  it('renders without crashing', () => {
    expect(toJson(shallow(<Game {...npr}/>))).toMatchSnapshot()
  })
})
