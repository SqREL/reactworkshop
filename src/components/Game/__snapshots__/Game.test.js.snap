// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders without crashing 1`] = `
<div
  className="c-game"
>
  <div
    className="c-game__img-wrapper"
  >
    <img
      className="c-game__img"
      src="http://google.com"
    />
  </div>
  <div
    className="c-game__content"
  >
    <div
      className="c-game__name"
    >
      test game name
    </div>
    <div
      className="c-game__price"
    >
      Price: 
      1488
    </div>
    <button
      className="o-button c-game__add-button"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      Add to cart
    </button>
  </div>
</div>
`;

exports[`Prce free renders without crashing 1`] = `
<div
  className="c-game"
>
  <div
    className="c-game__img-wrapper"
  >
    <img
      className="c-game__img"
      src="http://google.com"
    />
  </div>
  <div
    className="c-game__content"
  >
    <div
      className="c-game__name"
    >
      test game name
    </div>
    <div
      className="c-game__price"
    >
      Price: 
      Free
    </div>
    <button
      className="o-button c-game__add-button"
      onClick={[MockFunction]}
    >
      Add to cart
    </button>
  </div>
</div>
`;
